From 1af34f6d6a0603e828ca11dfe7b6053ef5436290 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Tue, 05 Jul 2022 16:41:53 +0300
Subject: [PATCH] [SW-88751] greco: fix greco card stuck on cold boot.

On some host it is observed that trying to access pcie aux
registers before prtsn assertion is causing the arc to get
hung. Hence access pcie aux registers after the prstn assertion.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: Ib9fb30ae6d5d0b40b60e17d80804b27636d4f6b7
---

diff --git a/arch/greco/common/common.c b/arch/greco/common/common.c
index 54360a4..263552d 100644
--- a/arch/greco/common/common.c
+++ b/arch/greco/common/common.c
@@ -698,6 +698,9 @@
 {
 	u32 val;
 
+	/* Wait for PRSTn before accessing pcie_aux registers */
+	while (!regmap_readl(mmPSOC_GLOBAL_CONF_PRSTN))
+		;
 	/*
 	 * If FLR reset is issued by FW, the reset
 	 * needs to be cleared by FW as the bit is not
